firmware/defmt-rtt/src/channel.rs

```diff
@@ -13,13 +10,13 @@ pub(crate) struct Channel {
     pub buffer: *mut u8,
     pub size: usize,
     /// Written by the target.
-    pub write: AtomicUsize,
+    pub write: critical_section::Mutex<RefCell<usize>>,
```

Comment:
Is there a benefit to the mutex approach instead of using the `atomic-polyfill` crate from `embassy`?
If I understand correctly, that crate will use the native atomics when preset, and polyfill atomics using `critical-section` when native ones aren't available

Reply:
That might indeed be the better option.

Reply:
See https://github.com/knurling-rs/defmt/pull/702